components/daikin_s21/climate/__init__.py

```diff
@@ -30,6 +31,7 @@
             cv.Optional(
                 CONF_SETPOINT_INTERVAL, default="300s"
             ): cv.positive_time_period_seconds,
+            cv.Optional(CONF_HAS_PRESETS, default=True): cv.boolean,
```

Comment:
Defaulting to `True` here is backwards-incompatible and will break existing configs for some units. For instance, my units do not support the F6/F7 queries (even though they do support powerful mode). As a result, building my existing configs with this change causes the unit to report incorrect data to Home Assistant, but setting this to false fixes the issue.

An alternative would be some sort of graceful degradation in the event presets do not work, though that would require some reworking of how the module elects which queries to use. This would be a worthwhile/valuable enhancement in general, as I suspect there are going to be more cases where different queries are required, but I don't expect that in this PR.